pkg/cmd/serve/serve.go

```diff
@@ -161,40 +148,25 @@ func serveRun(ctx context.Context, cmd *cobra.Command, args []string, datastoreO
 		[]float64{.006, .010, .018, .024, .032, .042, .056, .075, .100, .178, .316, .562, 1.000},
 	))
 
-	middleware := grpc.ChainUnaryInterceptor(
-		requestid.UnaryServerInterceptor(requestid.GenerateIfMissing(true)),
-		logmw.UnaryServerInterceptor(logmw.ExtractMetadataField("x-request-id", "requestID")),
-		grpclog.UnaryServerInterceptor(grpczerolog.InterceptorLogger(log.Logger)),
-		otelgrpc.UnaryServerInterceptor(),
-		grpcauth.UnaryServerInterceptor(auth.RequirePresharedKey(token)),
-		grpcprom.UnaryServerInterceptor,
-		servicespecific.UnaryServerInterceptor,
-	)
+	if len(config.UnaryMiddleware) == 0 || len(config.StreamingMiddleware) == 0 {
+		config.UnaryMiddleware, config.StreamingMiddleware = cmdutil.DefaultMiddleware(log.Logger, config.PresharedKey)
+	}
 
-	streamMiddleware := grpc.ChainStreamInterceptor(
-		requestid.StreamServerInterceptor(requestid.GenerateIfMissing(true)),
-		logmw.StreamServerInterceptor(logmw.ExtractMetadataField("x-request-id", "requestID")),
-		grpclog.StreamServerInterceptor(grpczerolog.InterceptorLogger(log.Logger)),
-		otelgrpc.StreamServerInterceptor(),
-		grpcauth.StreamServerInterceptor(auth.RequirePresharedKey(token)),
-		grpcprom.StreamServerInterceptor,
-		servicespecific.StreamServerInterceptor,
-	)
+	// TODO: should we validate that there is a psk-check middleware if a psk is set?
```

Comment:
This sounds like a good thing to have before shipping this.

Reply:
I wasn't totally sure this was a good idea, similar to your other comment ("What if you really don't want any middleware?") 

For example, if we want to provide other authn options in the future, the psk might be superseded by some other auth middleware.

Reply:
I think at that point in time we'd just have to change the checking code.

`WithAuthentication(presharedkeyMiddleware(key))` might be a way forward and we can change the strategies passed to WithAuthentication in the future.